### PR TITLE
Fixed parsing of subscribed topics containing slashes.

### PR DESCRIPTION
**Pull request by frizzy**

### CODE REVIEW

**Comment by henrikekblad** [comment]
Jenkins, build examples please.

File: core/MyProtocolMySensors.cpp [hunk 1]
```diff
@@ -120,40 +120,42 @@ bool protocolMQTTParse(MyMessage &message, char* topic, uint8_t* payload, unsign
...
-	for (str = strtok_r(topic, "/", &p); str && i <= 5;
+  if (topic != strstr(topic, MY_MQTT_SUBSCRIBE_TOPIC_PREFIX)) {
+		// Prefix doesn't match incoming topic
+		return false;
+	}
+	// Extract the topic sans prefix
+	strncpy(topicTrimmed, &topic[strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + 1], topiclen);
+  topicTrimmed[topiclen] = (char)0;
+
+	for (str = strtok_r(topicTrimmed, "/", &p); str && i <= 5;
```
Comment (henrikekblad) [inline core/MyProtocolMySensors.cpp]: Do you really need to copy topic into the new buffer topicTrimmed?

I _think_ you just could add to the topic pointer this:

for (str = strtok_r(topic + strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX), "/", &p);

Maybe you could test this in your environment?

Comment (frizzy) [inline core/MyProtocolMySensors.cpp]: Great! That looks way more elegant. I'll try that on my environment today. Thanks.

**Comment by henrikekblad** [comment]
Thanks @frizzy!

**Pull request merged**